### ISSUE TITLE
Error while storing scores

### ISSUE THREAD

**Issue by alecharp** [jenkins-infra/plugin-health-scoring#175]
While looking at #174, we can see in the logs a constraint violation while storing the score:

```
ERROR: duplicate key value violates unique constraint "scores_pkey"
```

```
org.springframework.dao.DataIntegrityViolationException: could not execute statement; SQL [n/a]; constraint [scores_pkey]
[...]
Caused by: org.postgresql.util.PSQLException: ERROR: duplicate key value violates unique constraint "scores_pkey"
```

**Comment by impact-maker** [comment]
@alecharp  Hi, I would like to work on this bug and make some valuable contribution. Is this issue about to get closed or needs some testing.

**Comment by alecharp** [comment]
I incorrectly linked this issue with another pull requests.

**Comment by impact-maker** [comment]
Hi @alecharp I have installed java, maven, node and yarn on my system but failed to run the mvn spring-boot:run -Dskip.yarn command. Could you help me figure out the reason, I reckon it might be due to the environment variables which is not clear from the contributing documentation. Alternatively can I create a beginner friendly issue and work on it to update the contributing documentation once I am able to build and run the project.

**Comment by impact-maker** [comment]
> I incorrectly linked this issue with another pull requests.

Hi @alecharp, Also just to confirm the status of this issue.

**Comment by alecharp** [comment]
> which is not clear from the contributing documentation

I'm sorry about that. Could you join https://gitter.im/jenkinsci/GSoC-Plugin_Health_Score so we can discuss it here? Just to keep this issue about the database problem. Thanks.

> Alternatively can I create a beginner friendly issue and work on it to update the contributing documentation once I am able to build and run the project.

This is an excellent idea! Please do so that we improve the contribution guide. This would be marvelous.

**Comment by alecharp** [comment]
I'm working back on this issue because it is blocking plugins scores from being updated.

**Comment by alecharp** [comment]
Thanks to @lemeurherve, I got an SQL dump of the database and I found something odd. 
The `scores.id` is set using the sequence `scores_seq`, which is incremented automatically. Normally. However, I can see that the current `nextval` of the sequence is `3451` when there is already a `3451` id. 

The `id` sequence seems to have to reset to `1` around the 6th of december of 2022, as they go from `345668` to `1` at that date.

**Comment by alecharp** [comment]
The time of the reset looks like it occurs with the release `1.3.1`, probably during its deployment. 
As the method to persist scores was not marked as transactional at this moment

https://github.com/jenkins-infra/plugin-health-scoring/blob/678b36e8320a43f190136b785c0c4c1edbf02dc8/src/main/java/io/jenkins/pluginhealth/scoring/service/ScoreService.java#L44-L46

It is possible that the deployment occurred during a insertion and the sequence was incorrectly managed. 

This was fixed in `1.6.1` with https://github.com/jenkins-infra/plugin-health-scoring/commit/4a69af7829ead5260a8bfffd1528f1553007a4f0

There are two potential solutions:
 - move ID of scores to UUID which would hide this kind of reset 
 - using a pooled low strategy for the id sequence

The current strategy for the sequence is `AUTO` which let use the "best" one for the provider (postgresql here). And it is the pooled low strategy.

However, in order to get out of the current situation, we need to clear the table from any entry and reset the sequence. 
This would be the same if we wanted to change to UUID for the primary key.

Next steps:
 - [ ] Ask to clear the `scores` table and reset the sequence nextval to 1.
